docs/dev/table/streaming/match_recognize.md

```diff
@@ -0,0 +1,654 @@
+
+for input:
+
+{% highlight text %}
+ symbol   tax   price.         rowtime
+======== ===== ======== =====================
+ XYZ      1     10       2018-09-17 10:00:02
+ XYZ      2     12       2018-09-17 10:00:03
+ XYZ      1     13       2018-09-17 10:00:04
+ XYZ      2     11       2018-09-17 10:00:05
+{% endhighlight %}
+
+will produce:
+
+{% highlight text %}
+ symbol   startPrice   topPrice   lastPrice
+======== ============ ========== ===========
+ XYZ      10           13         11
+{% endhighlight %}
+
+Event stream navigation
+------------------
+
+### Pattern variable reference
+Pattern variable reference e.g. A.price describes a set of rows mapped so far to A plus current row, if we try to match current row to A. If the expression requires a single row e.g. A.price > 10, A.price selects the last value of the row.
+
+If no pattern variable is specified e.g. SUM(price) it references the default pattern variable which references all variables in the pattern, so it constitutes row set of all rows mapped so far plus current row.
+
+Example:
+
+{% highlight sql %}
+...
+PATTERN (A B+)
+DEFINE
+  A as A.price > 10,
+  B as B.price > A.price AND SUM(price) < 100 AND SUM(B.price) < 80
+{% endhighlight %}
+  
+By `{A.price}` we describe set of rows that the values in DEFINE clause are evaluated on. The rows are referenced by its number
+
+<table class="table table-bordered">
+  <thead>
+    <tr>
+      <th>no</th>
+      <th>price</th>
+      <th>CLAS</th>
+      <th>{A.price}</th>
+      <th>{B.price}</th>
+      <th>{price}</th>
+      <th>A.price</th>
+      <th>B.price</th>
+      <th>SUM(price)</th>
+      <th>SUM(B.price)</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>1</td>
+      <td>10</td>
+      <td>-&gt; A</td>
+      <td>1</td>
+      <td>-</td>
+      <td>-</td>
+      <td>10</td>
+      <td>-</td>
+      <td>-</td>
+      <td>-</td>
+    </tr>
+    <tr>
+      <td>2</td>
+      <td>15</td>
+      <td>-&gt; B</td>
+      <td>1</td>
+      <td>2</td>
+      <td>1,2</td>
+      <td>10</td>
+      <td>15</td>
+      <td>25</td>
+      <td>15</td>
+    </tr>
+    <tr>
+      <td>3</td>
+      <td>20</td>
+      <td>-&gt; B</td>
+      <td>1</td>
+      <td>2,3</td>
+      <td>1,2,3</td>
+      <td>10</td>
+      <td>20</td>
+      <td>45</td>
+      <td>35</td>
+    </tr>
+    <tr>
+      <td>4</td>
+      <td>31</td>
+      <td>-&gt; B</td>
+      <td>1</td>
+      <td>2,3,4</td>
+      <td>1,2,3,4</td>
+      <td>10</td>
+      <td>31</td>
+      <td>76</td>
+      <td>66</td>
+    </tr>
+    <tr>
+      <td>5</td>
+      <td>35</td>
+      <td></td>
+      <td>1</td>
+      <td>2,3,4,5</td>
+      <td>1,2,3,4,5</td>
+      <td>10</td>
+      <td>35</td>
+      <td>111</td>
+      <td>101</td>
+    </tr>
+  </tbody>
+</table>
+
+### Pattern variable indexing
+
+Apache Flink allows navigating within events that were mapped to a particular pattern variable with so called logical offsets. This can be expressed
```

Comment:
Mention that this only makes sense when using `+`, `*` etc.

Reply:
Actually it makes sense in other cases as well:
* for the default pattern variable `LAST(price, 2)`, which indexes all events mapped so far 
* will make even more sense once we fully support `SUBSET`

Reply:
You are right. Maybe you can add your example as well in this section.